pkg/operator/controller/status/controller.go

```diff
@@ -313,8 +331,24 @@ func checkAllIngressesAvailable(ingresses []operatorv1.IngressController) bool {
 	return len(ingresses) != 0
 }
 
-// computeOperatorDegradedCondition computes the operator's current Degraded status state.
-func computeOperatorDegradedCondition(ingresses []operatorv1.IngressController) configv1.ClusterOperatorStatusCondition {
+func computeOperatorDegradedCondition(ingresses []operatorv1.IngressController, unmanagedGatewayAPICRDNames string) configv1.ClusterOperatorStatusCondition {
+	condition := computeIngressControllerDegradedCondition(ingresses)
+	if len(unmanagedGatewayAPICRDNames) > 0 {
+		gwapiCRDMsg := fmt.Sprintf("Unmanaged Gateway API CRDs found: %s.", unmanagedGatewayAPICRDNames)
+		if condition.Status != configv1.ConditionTrue {
+			condition.Status = configv1.ConditionTrue
+			condition.Reason = "GatewayAPICRDsDegraded"
+			condition.Message = gwapiCRDMsg
+		} else {
+			condition.Reason = "MultipleComponentsDegraded"
+			condition.Message = condition.Message + gwapiCRDMsg
```

Comment:
This is getting really spaghetti-like.  The code should be restructured to compute a series of inputs that get aggregated to compute the status condition.  `computeIngressUpgradeableCondition` is a really nice example of this.  In the interest of time, we can leave that as a follow-up.

Reply:
release-4.18's `computeOperatorUpgradeableCondition` follows a similar pattern, but it could be made a little more generic instead of having a long `if`/`else if` chain.  `computeIngressUpgradeableCondition` just collects a bunch of error values and uses `kerrors.NewAggregate` at the end.

Reply:
@Miciah : can you please check out [the new approach](https://github.com/openshift/cluster-ingress-operator/blob/d9b445d32195572049659a162060fd77af10338b/pkg/operator/controller/status/controller.go#L406-L419) to the computation of the Degraded status?